tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js

```diff
@@ -1,25 +1,283 @@
+        .click(selector)
+        .click('@dateTimeOkButton')
+    },
+    /**
+     * sets provided day in expiry date field on webUI
+     *
+     * @param selector selector for expiry date "day"
+     * @returns {Promise<void>}
+     */
+    setExpiryDateDay: async function (selector) {
+      await this
+        .waitForElementVisible(selector)
+        .click(selector)
+        .click('@dateTimeOkButton')
+    },
+    setPublicLinkRole: async function (value) {
+      const roleSelectorXpath = util.format(this.elements.roleButton.selector, value)
+      const roleSelector = { locateStrategy: 'xpath', selector: roleSelectorXpath }
+      await this
+        .waitForElementVisible(roleSelector)
+        .click(roleSelector)
+    },
+    setPublicLinkName: async function (value) {
+      await this
+        .waitForElementVisible('@publicLinkNameInputField')
+        .clearValue('@publicLinkNameInputField')
+        .setValue('@publicLinkNameInputField', value)
+    },
+    setPublicLinkPassword: async function (value) {
+      await this
+        .waitForElementVisible('@publicLinkPasswordField')
+        .clearValue('@publicLinkPasswordField')
+        .setValue('@publicLinkPasswordField', value)
+    },
+    setPublicLinkExpiryDate: async function (value) {
+      const date = value.split(/-/)
+      const yearSelector = this.setExpiryDateYearSelectorXpath(date[0])
+      const monthSelector = this.setExpiryDateMonthSelectorXpath(date[1])
+      const daySelector = this.setExpiryDateDaySelectorXpath(date[2])
+      await this
+        .initAjaxCounters()
+        .waitForElementVisible('@linkExpirationDateField')
+        .click('@linkExpirationDateField')
+      await this.setExpiryDateYear(yearSelector)
+      await this.setExpiryDateMonth(monthSelector)
+      await this.setExpiryDateDay(daySelector)
+    },
+    /**
+     * function sets different fields for public link
+     *
+     * @param key fields like name, password, expireDate, role
+     * @param value values for the different fields to be set
+     * @returns {Promise<void>}
+     */
+    setPublicLinkForm: async function (key, value) {
+      if (key === 'role') {
+        await this.setPublicLinkRole(value)
+      } else if (key === 'name') {
+        await this.setPublicLinkName(value)
+      } else if (key === 'password') {
+        await this.setPublicLinkPassword(value)
+      } else if (key === 'expireDate') {
+        await this.setPublicLinkExpiryDate(value)
+      }
+    },
+    /**
+     * checks if the provided expiryDate is past date or not
+     *
+     * @param linkName name of the public link
+     * @param pastDate provided past date for inspection
+     *  pastDate should be in form 2000-August-7 | 2000-Aug-7
+     *  leading zeros before day are removed
+     * @returns {Promise<void>}
+     */
+    assertDisabledExpiryDate: async function (linkName, pastDate) {
+      this.clickLinkEditBtn(linkName)
+      const pastDay = pastDate.split(/-/)
```

Comment:
Spread the array as follows:
```suggestion
      const [year, month, day] = pastDate.split('-')
```